### ISSUE TITLE
Huggingface GGUF loader missing

### ISSUE THREAD

**Issue by allo-** [oobabooga/text-generation-webui#6905]
### Describe the bug

I've upgraded to the latest git version and now I don't see the huggingface loader for gguf models and the model list contains Model-HF/Model.gguf instead of Model-HF.

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Reproduction

- Check out current main branch
- Install a fresh virtual environment from requirements/full/requirements.txt
- Run the webui and click on Models
- There is no huggingface loader for gguf and the create huggingface model tab on the right is also missing

### Logs

```shell
-
```

### System Info

```shell
Linux, torch 2.7
```

**Comment by allo-** [comment]
It looks like it was maybe lost during the llama.cpp loader rework?

I was using it to have some samplers that pure llama.cpp does not support, so it would be good to be able to have it or to have another option for sampler hijack to work with gguf models.

**Comment by allo-** [comment]
Looks like #6846 removed it. Is there any way to get custom logits processors working on gguf with the new loader? It looks like it is currently restricted to what llama.cpp supports.

**Comment by robertgro** [comment]
Please re-introduce this.

**Comment by kuronekosaiko** [comment]
If your main goal is to use custom logits processors/samplers with recent GGUF models, and you're okay with forgoing features from the very latest text-generation-webui (TGWUI) versions, you can achieve this by using TGWUI v2.7 (which retains the `llamacpp_hf` loader) combined with an up-to-date `llama-cpp-python` build.

This approach allows `llamacpp_hf`'s custom logits processing capabilities while ensuring compatibility with the newest GGUF model formats.

Here's a streamlined guide:

1.  **Revert TGWUI to v2.7:**
    *   In your `text-generation-webui` git repository:
        ```bash
        git fetch --all --tags
        git checkout v2.7
        ```
    *   Alternatively, download the v2.7 release directly from GitHub.

2.  **Build/Obtain latest `llama-cpp-python` wheels:**
    *   Fork `oobabooga/llama-cpp-python-cuBLAS-wheels` on GitHub.
    *   In your fork, navigate to the 'Actions' tab and run the 'Build Everything TGW' workflow. This will take ~2 hours.

3.  **Update TGWUI v2.7 requirements:**
    *   Once the workflow completes, find the release created in your forked `llama-cpp-python-cuBLAS-wheels` repository. Copy the direct download link(s) for the `.whl` file(s) you need (e.g., appropriate CUDA version or CPU).
    *   In your TGWUI v2.7 directory, edit the `requirements.txt` (and `requirements_cuda.txt` if using CUDA).
    *   Locate the lines specifying `llama_cpp_python` or `llama_cpp_python_cuda` (which will point to `oobabooga/llama-cpp-python-cuBLAS-wheels`). Replace the original wheel URL after the `@` symbol with the URL(s) to the wheel(s) from *your* fork's release assets.

After these steps, run the TGWUI setup script. It will install the newer `llama-cpp-python` from your specified URL. You'll then be able to use the `llamacpp_hf` loader in TGWUI v2.7 with custom logits processors on the latest GGUF models.

The trade-off is that you'll be on TGWUI v2.7 and won't have features or fixes introduced in later TGWUI versions.

**Comment by allo-** [comment]
Thanks, I will look into it.

I had a proof of concept for a new sampler implemented as LogitsProcessor in text-generation-webui. I also may try to reimplement it in llama.cpp directly it was just much faster to do with a small python class.

**Comment by kuronekosaiko** [comment]
I'm personally using the method to run the latest qwen3 with custom samplers. I'd be interested in seeing your new sampler implemented as a LogitsProcessor in text-generation-webui.

**Comment by allo-** [comment]
I wanted to do a pull request when I am satisfied with it, then the support for the HF loader got pulled. The incomplete part is that it currently ignores the batching dimension and only works with the first request and maybe is a bit crude about decoding to a string which may be less efficient than one might get with pure torch.

I am also not sure if I first want to write up the theory completely. It could be a short paper, but I am currently not having time to do a full write-up with comparisons and other things people like to read in addition to the math and algorithm.
